datadog/resource_datadog_synthetics_test_.go

```diff
@@ -1396,3 +1588,153 @@ func getCertificateStateValue(content string) string {
+		return []string{"check", "element", "value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_ASSERT_ELEMENT_PRESENT:
+		return []string{"element"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_ASSERT_EMAIL:
+		return []string{"email"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_ASSERT_FILE_DOWNLOAD:
+		return []string{"file"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_ASSERT_FROM_JAVASCRIPT:
+		return []string{"code", "element"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_ASSERT_PAGE_CONTAINS:
+		return []string{"value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_ASSERT_PAGE_LACKS:
+		return []string{"value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_CLICK:
+		return []string{"click_type", "element"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_EXTRACT_FROM_JAVASCRIPT:
+		return []string{"code", "element", "variable"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_EXTRACT_VARIABLE:
+		return []string{"element", "variable"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_GO_TO_EMAIL_LINK:
+		return []string{"value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_GO_TO_URL:
+		return []string{"value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_HOVER:
+		return []string{"element"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_PLAY_SUB_TEST:
+		return []string{"playing_tab_id", "subtest_public_id"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_PRESS_KEY:
+		return []string{"modifiers", "value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_REFRESH:
+		return []string{}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_RUN_API_TEST:
+		return []string{"request"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_SCROLL:
+		return []string{"element", "x", "y"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_SELECT_OPTION:
+		return []string{"element", "value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_TYPE_TEXT:
+		return []string{"delay", "element", "value"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_UPLOAD_FILES:
+		return []string{"element", "files", "with_click"}
+
+	case datadogV1.SYNTHETICSSTEPTYPE_WAIT:
+		return []string{"value"}
+	}
+
+	return []string{}
+}
+
+func convertStepParamsValueForConfig(stepType datadogV1.SyntheticsStepType, key string, value interface{}) interface{} {
+	switch key {
+	case "element", "email", "file", "request":
+		result := make(map[string]interface{})
+		getMetadataFromJSON([]byte(value.(string)), &result)
+		return result
+
+	case "playing_tab_id":
+		result, _ := strconv.Atoi(value.(string))
+		return result
+
+	case "value":
+		if stepType == datadogV1.SYNTHETICSSTEPTYPE_WAIT {
+			result, _ := strconv.Atoi(value.(string))
+			return result
+		}
+
+		return value
+
+	case "variable":
+		return value.([]interface{})[0]
+	}
+
+	return value
+}
+
+func convertStepParamsValueForState(key string, value interface{}) interface{} {
+	switch key {
+	case "element", "email", "file", "request":
+		result, _ := json.Marshal(value)
+		return string(result)
+
+	case "playing_tab_id", "value":
+		return convertToString(value)
+
+	case "variable":
+		return []interface{}{value}
+	}
+
+	return value
+}
+
+func convertStepParamsKey(key string) string {
```

Comment:
It seems a bit weird to have a conversion method which handle both sides, would it make sense to divide between two methods (something like `toSnakeCase`/`toCamelCase`, or `toState`/`toConfig`)?

Reply:
This was to avoid having to create the `toSnakeCase` / `toCamelCase` methods but I can do that 👍